### PR TITLE
Update cypress

### PR DESCRIPTION
**Pull request by JaviIG**

### CODE REVIEW

File: packages/x-components/tests/support/index.ts [hunk 1]
```diff
@@ -195,25 +199,6 @@ const customCommands: CustomCommands = {
     cy.getByDataTest('loading').should('exist');
     cy.getByDataTest('loading').should('not.exist').then(noOp);
   },
-  fakeSearchResponse: searchResponse => {
```
Comment (JaviIG) [inline packages/x-components/tests/support/index.ts:198]: This was unused legacy code. Now we have a new e22 adapter.

File: packages/x-components/tests/support/index.ts [hunk 2]
```diff
@@ -235,10 +220,5 @@ const customDualCommands: AddPreviousParam<CustomDualCommands> = {
...
 
 // Register the commands
-forEach(customCommands, (name, implementation) => {
-  Cypress.Commands.add(name, implementation);
-});
-
-forEach(customDualCommands, (name, implementation) => {
-  Cypress.Commands.add(name, { prevSubject: 'optional' }, implementation);
-});
+Cypress.Commands.addAll(customCommands);
```
Comment (JaviIG) [inline packages/x-components/tests/support/index.ts:223]: I couldn't say no to this easy change using Cypress API

File: packages/x-components/tests/unit/main-scroll.spec.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ function renderMainScroll({
 
         html,
         body,
-        #__cy_root {
+        [data-cy-root] {
```
Comment (JaviIG) [inline packages/x-components/tests/unit/main-scroll.spec.ts:86]: Cypress root element selector changed

File: packages/x-components/tests/unit/main-scroll.spec.ts [hunk 4]
```diff
@@ -307,5 +307,5 @@ interface RenderMainScrollAPI {
    */
   getScrollingElement: () => Cypress.Chainable<JQuery>;
   /** Spy of {@link ScrollXEvents.UserScrolledToElement}. */
-  userScrolledToElementSpy: () => Cypress.Chainable<Cypress.SinonSpyAgent<any>>;
+  userScrolledToElementSpy: () => Cypress.Chainable<ReturnType<typeof cy.spy>>;
```
Comment (JaviIG) [inline packages/x-components/tests/unit/main-scroll.spec.ts:310]: I don't want to fight  Sinon types, which isn't even a dependency of us

File: packages/x-components/tests/support/index.ts [hunk 5]
```diff
@@ -14,8 +12,9 @@ import Loggable = Cypress.Loggable;
 import Shadow = Cypress.Shadow;
 import Timeoutable = Cypress.Timeoutable;
 import Withinable = Cypress.Withinable;
+import CommandFns = Cypress.CommandFns;
 
-interface CustomCommands {
+interface CustomCommands extends CommandFns {
```
Comment (JaviIG) [inline packages/x-components/tests/support/index.ts:17]: Extending the CommandFns interface so we ensure all functions are compatible commands.

File: packages/x-components/package.json [hunk 6]
```diff
@@ -79,8 +79,8 @@
     "vuex": "^3.0.0"
   },
   "devDependencies": {
-    "@cypress/vue": "~2.2.4",
-    "@cypress/webpack-dev-server": "~1.8.4",
+    "@badeball/cypress-cucumber-preprocessor": "~13.0.1",
+    "@bahmutov/cypress-esbuild-preprocessor": "~2.1.4",
```
Comment (JaviIG) [inline packages/x-components/package.json:83]: cumpress-cucumber-preprocessor has been abandoned and [transferred to badeball](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689). Fortunately, there were just a few breaking changes. The `And` command does not exist anymore, the imports path has changed.

File: packages/x-components/package.json [hunk 7]
```diff
@@ -98,9 +98,7 @@
     "@vue/test-utils": "~1.0.3",
     "@vue/vue2-jest": "~27.0.0-alpha.3",
     "autoprefixer": "~10.4.4",
-    "cypress": "~8.4.0",
-    "cypress-cucumber-preprocessor": "~4.0.0",
-    "cypress-plugin-tab": "~1.0.5",
```
Comment (JaviIG) [inline packages/x-components/package.json:103]: Tab plugin was not used.

File: packages/x-components/package.json [hunk 8]
```diff
@@ -79,8 +79,8 @@
     "vuex": "^3.0.0"
   },
   "devDependencies": {
-    "@cypress/vue": "~2.2.4",
-    "@cypress/webpack-dev-server": "~1.8.4",
```
Comment (JaviIG) [inline packages/x-components/package.json:83]: These two are not needed anymore. They arenow included inside cypress.

File: packages/x-components/cypress.config.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    retries: {
+      openMode: 0,
+      runMode: 1
+    },
+    async setupNodeEvents(on, config) {
+      await addCucumberPreprocessorPlugin(on, config);
+      on(
+        'file:preprocessor',
+        createBundler({
+          plugins: [createEsbuildPlugin(config)]
```
Comment (JaviIG) [inline packages/x-components/cypress.config.ts:26]: The recommended option  with the new cypress-cucumber-preiprocessor is to use esbuild (I won't complain).

**Review by mavmaf — APPROVED** [review]
<3
we might want to add as well Webkit support. Well, at least I know I want to hahaha
https://docs.cypress.io/guides/guides/launching-browsers#WebKit-Experimental

**Comment by tiborux** [comment]
Can you update the pr solving conflicts? @JaviIG

**Pull request closed without merging**